[2025-03-17T01:56:38.224+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-03-17T01:56:36.445129+00:00 [queued]>
[2025-03-17T01:56:38.231+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-03-17T01:56:36.445129+00:00 [queued]>
[2025-03-17T01:56:38.231+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2025-03-17T01:56:38.238+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): reddit_extraction> on 2025-03-17 01:56:36.445129+00:00
[2025-03-17T01:56:38.245+0000] {standard_task_runner.py:57} INFO - Started process 50 to run task
[2025-03-17T01:56:38.247+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'etl_reddit_pipeline', 'reddit_extraction', 'manual__2025-03-17T01:56:36.445129+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmp67yndt79']
[2025-03-17T01:56:38.248+0000] {standard_task_runner.py:85} INFO - Job 2: Subtask reddit_extraction
[2025-03-17T01:56:38.304+0000] {task_command.py:415} INFO - Running <TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-03-17T01:56:36.445129+00:00 [running]> on host 7e74b270c79c
[2025-03-17T01:56:38.359+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='vijay Mopuru' AIRFLOW_CTX_DAG_ID='etl_reddit_pipeline' AIRFLOW_CTX_TASK_ID='reddit_extraction' AIRFLOW_CTX_EXECUTION_DATE='2025-03-17T01:56:36.445129+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-17T01:56:36.445129+00:00'
[2025-03-17T01:56:38.467+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/reddit_dag.py", line 11, in reddit_pipeline
    reddit = praw.Reddit(
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/util/deprecate_args.py", line 46, in wrapped
    return func(**dict(zip(_old_args, args)), **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/reddit.py", line 259, in __init__
    raise MissingRequiredAttributeException(
praw.exceptions.MissingRequiredAttributeException: Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.
[2025-03-17T01:56:38.476+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_reddit_pipeline, task_id=reddit_extraction, execution_date=20250317T015636, start_date=20250317T015638, end_date=20250317T015638
[2025-03-17T01:56:38.486+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 2 for task reddit_extraction (Required configuration setting 'client_id' missing. 
This setting can be provided in a praw.ini file, as a keyword argument to the Reddit class constructor, or as an environment variable.; 50)
[2025-03-17T01:56:38.511+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-03-17T01:56:38.526+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
